test/specs/reporters/summary.js

```diff
@@ -0,0 +1,73 @@
+var assert = require('assert');
+var sinon = require('sinon');
+
+var Checker = require('../../../lib/checker');
+var summaryReporter = require('../../../lib/reporters/summary');
+
+describe('reporters/summary', function() {
+    var checker;
+
+    beforeEach(function() {
+        checker = new Checker();
+        checker.registerDefaultRules();
+        checker.configure({ disallowKeywords: ['with'] });
+
+        sinon.stub(console, 'log');
+    });
+
+    afterEach(function() {
+        console.log.restore();
+    });
+
+    it('should correctly reports no errors', function() {
+        summaryReporter([checker.checkString('a++;')]);
+        assert.equal(console.log.getCall(0).args[0], 'No code style errors found.');
+        assert(console.log.calledOnce);
+    });
+
+    it('should correctly report 1 error', function() {
+        var expectedByFile = [
```

Comment:
Here, it seems you testing, although quite elegantly, logic of the external tool?